rr_arduino/rr_arduino.ino

```diff
@@ -174,3 +181,29 @@ void get_battery_state(long current_time){
     prev_time = millis();
   }
 }
+
+void raw_to_odom(double vel, int str_angle) {
+    long current_time = millis();
+    long time_diff = current_time - prev_time;
+    // If the robot is at the origin, calculate the position using steering angle and the velocity
+    if (str_angle != 1500 && x == 0.0 && y == 0.0) {
+        // Mapping str_angle integer to angle in degrees
+        theta = (str_angle - 1000) / (2000 - 1000) * 60 - 30;
+        x = vel * cos(theta * M_PI / 180.0) * time_diff;
```

Comment:
your kinematic model is missing a few parameters, try following the equations in the blog you mentioned earlier. https://nabinsharma.wordpress.com/2014/01/02/kinematics-of-a-robot-bicycle-model/

Reply:
I need the length of the car (L) to calculate the positions. I just put 0 as a placeholder.